src/pages/Home.tsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react'
+import { db as firebaseDB } from '../services/firebase-client'
+
+interface User {
+  username: string
+}
+
+export const HomePage = () => {
+  const [username, setUsername] = useState<string>("")
+  const [users, setUsers] = useState<User[]>([])
+
+  const handleInputChange = (event: any) => {
+    setUsername(event.target.value)
+  }
+  const createUser = async () => {
+    await firebaseDB.append('users', {
+      username: username
+    })
+
+    fetchUsers()
```

Comment:
I think this `createUser` calls `fetchUsers` to refresh UI is violating Single Responsibility Principal

Reply:
how about I create a function called `handleButtonClick` to create user and do the refresh UI? so `createUser` can keep its single responsibility of creating user on db

Reply:
`handleButtonClick` is a bad naming.
If you are adding multiple buttons in to the same page => are you going to name them
- `handleButton2Click`
- `handleButton3Click`
The naming should be simple, concise, and future proof